pkg/plugins/policies/meshhttproute/xds/filters/util.go

```diff
@@ -0,0 +1,14 @@
+package filters
+
+import (
+	envoy_route "github.com/envoyproxy/go-control-plane/envoy/config/route/v3"
+	"github.com/pkg/errors"
+)
+
+func UpdateRouteAction(envoyRoute *envoy_route.Route, updFunc func(*envoy_route.RouteAction) error) error {
```

Comment:
Why not just call `envoyRoute.Action.(*envoy_route.Route_Route)` inline and let the generation panic and fail?

Reply:
I think returning errors is better handling in this case. Configurer interface implies you can return an `error`, so why not do it? 

`URLRewriteConfigurer` and `RequestMirrorConfigurer` expect to find `envoyRoute` in state X (with RouteAction), but they find it in state Y (without RouteAction), so they just return an error `expect X, got Y`. 

And it's more developer friendly when you write unit tests for MeshHTTPRoute, instead of panic you have failed assertion:
```go
Expect(plugin.Apply(resourceSet, given.xdsContext, &given.proxy)).To(Succeed())
```

Reply:
I might be wrong about this case but wouldn't calling `updateRouteAction` with a `Route` that we just created that doesn't have an `.Action` set be a case of developer error and not runtime error? I don't think the user can configure Kuma in such a way to cause this error. It can only happen during development when we've incorrectly strung these configurers together while adding a feature, right?

It's only an issue because we're using this `Configurer` interface which takes an entire `Route` and not just a `RouteAction` directly. Without the interface we'd simply call `configureURLRewrite(*filter.URLRewrite, match.prefixMatch, &envoyRouteAction)` using the route we just unconditionally created 10 lines before. It's the only reason we're confident about calling `.Configure(NewURLRewrite...)` in the first place after all. It's not a state we can actually enter, it's just an artifact of the `Configurer` pattern.

What do we gain from seeing `error` as a return value then? IMO it's always better to limit the number of states our code can be in if possible. Whether `error` or `panic`, the effect must be the same, I think, that config generation for this `Dataplane` must fail. The code is so broken in that case that we can't recover and at least with `panic` we fail fast.

> And it's more developer friendly when you write unit tests for MeshHTTPRoute, instead of panic you have failed assertion

Not sure, if there were no `error` value, you would just write:

```
plugin.Apply(resourceSet, given.xdsContext, &given.proxy)

Expect(resourceSet[0]....).To(...)
```

and the test would fail because it called `panic("expected to have routeAction")`.

Reply:
But yeah, not a blocker for this PR.

Reply:
Not a blocker I guess, just an effect of the `Configurer` pattern, but it's noisy and makes it hard to know what the preconditions for using `NewURLRewrite` `Configurer` are as opposed to a normal Go function like `configureURLRewrite`.

Reply:
Yes, you are right it's an effect of the `Configurer` pattern. We can discuss later today what others think about it. I'm resolving this for now